### PR TITLE
Add support for StoreCursor to Hybrid SDK

### PR DESCRIPTION
**Pull request by jhorstsf**
@kchitalia @smcnulty-sfdc @wpeter-sfdc

### CODE REVIEW

File: SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs [hunk 1]
```diff
@@ -232,5 +238,75 @@ public static string GetSoupTableName(long soupId)
...
+                _cursors.Remove(id);
+            }
+        }
+
+        /// <summary>
+        /// Native implementation of pgMoveCursorToPageIndex
+        /// </summary>
+        public string MoveCursorToPageIndex(int id, int index)
+        {
+            if (!_cursors.ContainsKey(id))
```
Comment (jhorstsf) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: @kchitalia should I be logging anything in these functions if something unexpected happens?

Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: Yep, makes sense to log exceptions here.

Comment (jhorstsf) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/SmartStore.cs]: Ok, added a new commit which throws an exception if the cursor does not exist

File: SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/StoreCursor.cs [hunk 2]
```diff
@@ -0,0 +1,62 @@
+﻿#region Copyright
+// 
+//  Copyright (c) 2015, salesforce.com, inc.
+//  All rights reserved.
+//  Company Confidential
+//  
+#endregion
```
Comment (kchitalia) [inline SalesforceSDK/Salesforce.SDK.Hybrid.SmartStore/Source/Store/StoreCursor.cs]: Can you please copy over the copyright section from one of the existing files.

**Comment by kchitalia** [comment]
Apart from the copyright comment, everything looks good.

**Comment by jhorstsf** [comment]
@kchitalia copyright added in latest commit

**Comment by bhariharan** [comment]
:+1:

**Pull request merged**